reconciler/types.go

```diff
@@ -94,15 +96,65 @@ type BatchOperations[Obj any] interface {
 	DeleteBatch(context.Context, statedb.ReadTxn, []BatchEntry[Obj])
 }
 
-type StatusKind string
+// StatusKind marks the state of status.
+//
+// This is a struct for legacy reasons. This used to string, but to save on space
+// this is now a uint8. By wrapping it into a struct we force the use-sites to
+// use StatusKind.String() and makes it impossible to do string(foo.Status.Kind).
+type StatusKind struct {
+	kind uint8
+}
 
-const (
-	StatusKindPending    StatusKind = "Pending"
-	StatusKindRefreshing StatusKind = "Refreshing"
-	StatusKindDone       StatusKind = "Done"
-	StatusKindError      StatusKind = "Error"
+var (
```

Comment:
Could be out of context here but this feels a little odd aka consts -> global variables. Not sure if applicable but perhaps we could have gone with StatusKind uint8 and generate the string equivalent of the consts via go:generate. 
Aka have the compiler generate the map vs manually curating `statusKindToString`

Reply:
I wish I could do that, but since we have existing code doing `string(foo.Status.Kind)` we cannot do `type StatusKind uint8` since `string(uint8(0))` compiles and we want to force the code to switch to `foo.Status.Kind.String()`. Maybe in the future once all use-sites have converted over? For now I'd like to err on the safe side.